django/utils/module_loading.py

```diff
@@ -13,6 +13,11 @@ def cached_import(module_path, class_name):
         and getattr(spec, "_initializing", False) is False
     ):
         module = import_module(module_path)
+    if not hasattr(module, class_name):
```

Comment:
`class_name` allowed to also be a submodule name is confusing -- since cached_import() doesn't appear to be documented, I _think_ we can take the opportunity to rename to something like `submodule_or_class`. Not sure of the best name.

An inline comment here would also be helpful IMO. Otherwise it's not clear that getattr'ing a submodule off the parent module only sometimes works (i.e. when submodule already explicitly imported).

Reply:
I guess we could also enhance it to take top-level modules and just skip the getattr call. Then it becomes module_or_class. Or maybe "attribute" is enough?

Reply:
I agree, with the intended functionality, the names are quite confusing. As this is my first PR, I am not too familiar with Django naming conventions. I think “attribute” would suffice, as that encapsulates all importable items.